### PR TITLE
feat(TDP-4992): add dataset name

### PR DESCRIPTION
**Pull request by ncomont**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4992

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-webapp/src/app/services/playground/playground-service.js [hunk 1]
```diff
@@ -119,9 +123,86 @@ export default function PlaygroundService(
...
+			bsStyle: 'link',
+			onClick: () => $timeout(() => StateService.setIsPreprationPickerVisible(true)),
+		}] : [];
+
+		const controls = state.playground.isReadOnly ? [] : [
+			{
+				id: 'playground-lookup-icon',
+				label: $translate.instant('DATAGRID_PARAMETERS_GEAR'),
+				icon: 'talend-cog',
+				displayMode: TYPE_ICON_TOGGLE,
```
[the remark this answers is not in the document]
Comment (ncomont) [reply dataprep-webapp/src/app/services/playground/playground-service.js]: types are not exported, I'll do a PR for that

File: dataprep-webapp/src/app/services/playground/playground-service.js [hunk 2]
```diff
@@ -119,9 +123,86 @@ export default function PlaygroundService(
...
+				tooltipLabel: $translate.instant('REDO_ICON_TOOLTIP'),
+				hideLabel: true,
+				bsStyle: 'link',
+				disabled: !HistoryService.canRedo(),
+				onClick: () => $timeout(HistoryService.redo),
+			},
+		];
+
+		return [
+			...actions,
```
[the remark this answers is not in the document]
Comment (ncomont) [reply dataprep-webapp/src/app/services/playground/playground-service.js]: for readability reasons : to not have multiple ternary operators in the return statement.

File: dataprep-webapp/src/app/components/playground/header/playground-header-controller.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+		this.state = state;
+		this.StateService = StateService;
+		this.HelpService = HelpService;
+		this.PlaygroundService = PlaygroundService;
+		this.HistoryService = HistoryService;
+		this.LookupService = LookupService;
+		this.PreviewService = PreviewService;
+		this.$state = $state;
+		this.$stateParams = $stateParams;
```
[the remark this answers is not in the document]
Comment (ncomont) [reply dataprep-webapp/src/app/components/playground/header/playground-header-controller.js]: good catch, thanks

File: dataprep-webapp/src/app/components/playground/header/playground-header.html [hunk 4]
```diff
@@ -21,87 +21,17 @@
...
-        </li>
-    </ul>
-</nav>
+<pure-sub-header-bar
+    class="playground-sub-header"
+    ng-attr-title="$ctrl.title"
+    sub-title="$ctrl.subtitle"
+    editable="!$ctrl.state.playground.isReadOnly"
+    on-go-back="$ctrl.onClose"
+    on-edit="$ctrl.toggleNameEditMode"
```
[the remark this answers is not in the document]
Comment (ncomont) [reply dataprep-webapp/src/app/components/playground/header/playground-header.html]: added on TDP side, but not yet taken into account by TUI. I'll do a PR for that.

File: dataprep-webapp/src/app/css/base/_base.scss [hunk 5]
```diff
@@ -23,6 +23,10 @@
 	outline: none;
 }
 
+.tc-icon-toggle:focus {
+	border-color: inherit;
```
Comment (frassinier) [inline dataprep-webapp/src/app/css/base/_base.scss]: Do we need that? If so, can we put it next to pureheaderbar component, in a scss file?

File: dataprep-webapp/src/app/services/playground/playground-service.js [hunk 6]
```diff
@@ -119,9 +123,114 @@ export default function PlaygroundService(
...
 		// parameters
 		changeDatasetParameters,
+
+		// subheader
+		_getPreparationChooserAction,
+		_getDatasetParametersAction,
+		_getLookupAction,
+		_getUndoAction,
+		_getRedoAction,
+		_toggleLookupPane,
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: `_getLookupToggleAction` instead to be consistent

File: dataprep-webapp/src/app/services/playground/playground-service.spec.js [hunk 7]
```diff
@@ -134,6 +142,12 @@ describe('Playground Service', () => {
 		$translateProvider.translations('en', {
 			PREPARATION: 'Preparation',
 			TALEND: 'TALEND',
+			PREVIEW: 'PREVIEW',
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.spec.js]: You could reuse i18n file content instead and test en at least ¯\_(ツ)_/¯

File: dataprep-webapp/src/app/services/playground/playground-service.spec.js [hunk 8]
```diff
@@ -171,6 +185,287 @@ describe('Playground Service', () => {
...
+				spyOn(StateService, 'toggleDatasetParameters').and.returnValue();
+				spyOn(PlaygroundService, '_toggleLookupPane').and.returnValue();
+				spyOn(StateService, 'setLookupVisibility').and.returnValue();
+				spyOn(StateService, 'setStepInEditionMode').and.returnValue();
+				spyOn(LookupService, 'initLookups').and.returnValue();
+				spyOn(HistoryService, 'undo').and.returnValue();
+				spyOn(HistoryService, 'redo').and.returnValue();
+			}));
+
+			// need help on that :
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.spec.js]: Need to be rework then :)

File: dataprep-webapp/src/app/components/playground/header/_playground-header.scss [hunk 9]
```diff
@@ -10,85 +10,31 @@
...
-				cursor: not-allowed;
-				opacity: 0.5;
-			}
-
-			> icon {
-				line-height: 1rem;
-			}
-		}
-
+	#help-history-redo,
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/playground/header/_playground-header.scss]: please add `#subheader-action-switch-to-current-state`

File: dataprep-webapp/src/app/components/history-control/history-control-directive.js [hunk 10]
```diff
@@ -1,52 +0,0 @@
...
-
-		controllerAs: 'historyCtrl',
-		link(scope) {
-			function historyListener(event) {
-                // CTRL+Z
-				if (event.keyCode === 90 && event.ctrlKey) {
-					HistoryService.undo();
-				}
-                // Ctrl+Y
-				else if (event.keyCode === 89 && event.ctrlKey) {
```
[the remark this answers is not in the document]
Comment (frassinier) [reply dataprep-webapp/src/app/components/history-control/history-control-directive.js]: ✅

File: dataprep-webapp/src/app/components/playground/_playground.scss [hunk 11]
```diff
@@ -30,7 +30,7 @@ $slide-action-background: $base-background-color;
 
 $navigation-border: 1px;
 $navigation-height: $navbar-height;
-$sub-header-height: $navbar-height;
+$sub-header-height: 70px;
```
[the remark this answers is not in the document]
Comment (frassinier) [reply dataprep-webapp/src/app/components/playground/_playground.scss]: ✅

File: dataprep-webapp/src/app/components/playground/header/_playground-header.scss [hunk 12]
```diff
@@ -10,85 +10,31 @@
...
 .playground-sub-header {
 	@include flex-row();
-	min-height: $sub-header-height;
-	height: $sub-header-height;
-	box-shadow: 0px 2px 5px rgba(0, 0, 0, 0.25);
-	padding: 0 15px;
-	z-index: 15;
+	position: relative;
+	box-shadow: 0 2px 5px rgba(0, 0, 0, 0.25);
+	z-index: 1;
```
[the remark this answers is not in the document]
Comment (frassinier) [reply dataprep-webapp/src/app/components/playground/header/_playground-header.scss]: ✅

File: dataprep-webapp/src/app/components/playground/header/_playground-header.scss (hunk 9 above)
Comment (frassinier) [reply dataprep-webapp/src/app/components/playground/header/_playground-header.scss]: ✅

File: dataprep-webapp/src/app/components/playground/header/playground-header-controller.js [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+		this.$translate = $translate;
+		this.$stateParams = $stateParams;
+
+		this.toggleNameEditMode = this.toggleNameEditMode.bind(this);
+		this.editName = this.editName.bind(this);
+		this.cancelEdit = this.cancelEdit.bind(this);
+	}
+
+	toggleNameEditMode() {
+		this.StateService.setNameEditionMode(true);
```
[the remark this answers is not in the document]
Comment (frassinier) [reply dataprep-webapp/src/app/components/playground/header/playground-header-controller.js]: ✅

File: dataprep-webapp/src/app/components/playground/header/playground-header-controller.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (frassinier) [reply dataprep-webapp/src/app/components/playground/header/playground-header-controller.js]: ✅

File: dataprep-webapp/src/app/components/playground/header/playground-header-controller.js [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+					this.$state.go(PLAYGROUND_PREPARATION_ROUTE, { prepid: id });
+				})
+				.finally(() => {
+					this.StateService.setNameEditionMode(false);
+					this.loading = false;
+				});
+		}
+	}
+
+	cancelEdit() {
```
[the remark this answers is not in the document]
Comment (frassinier) [reply dataprep-webapp/src/app/components/playground/header/playground-header-controller.js]: ✅

File: dataprep-webapp/src/app/components/playground/header/playground-header.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (frassinier) [reply dataprep-webapp/src/app/components/playground/header/playground-header.html]: Added on Talend/ui ✅

File: dataprep-webapp/src/app/css/base/_base.scss (hunk 5 above)
Comment (frassinier) [reply dataprep-webapp/src/app/css/base/_base.scss]: ✅

File: dataprep-webapp/src/app/services/playground/playground-service.spec.js (hunk 7 above)
Comment (frassinier) [reply dataprep-webapp/src/app/services/playground/playground-service.spec.js]: ✅

File: dataprep-webapp/src/app/services/playground/playground-service.spec.js (hunk 8 above)
Comment (frassinier) [reply dataprep-webapp/src/app/services/playground/playground-service.spec.js]: ✅

**Pull request merged**